src/hera/workflows/cluster_workflow_template.py

```diff
@@ -0,0 +1,70 @@
+        assert self.namespace, "workflow namespace not defined"
+        return self.workflows_service.create_cluster_workflow_template(
+            ClusterWorkflowTemplateCreateRequest(template=self.build())
+        )
+
+    def get(self) -> TWorkflow:
+        """Attempts to get a workflow template based on the parameters of this template e.g. name + namespace"""
+        assert self.workflows_service, "workflow service not initialized"
+        assert self.namespace, "workflow namespace not defined"
+        assert self.name, "workflow name not defined"
+        return self.workflows_service.get_cluster_workflow_template(name=self.name)
+
+    def update(self) -> TWorkflow:
+        """
+        Attempts to perform a workflow template update based on the parameters of this template
+        e.g. name, namespace. Note that this creates the template if it does not exist. In addition, this performs
+        a get prior to updating to get the resource version to update in the first place. If you know the template
+        does not exist ahead of time, it is more efficient to use `create()` directly to avoid one round trip.
+        """
+        assert self.workflows_service, "workflow service not initialized"
+        assert self.namespace, "workflow namespace not defined"
+        assert self.name, "workflow name not defined"
+        # we always need to do a get prior to updating to get the resource version to update in the first place
+        # https://github.com/argoproj/argo-workflows/pull/5465#discussion_r597797052
+
+        template = self.build()
+        try:
+            curr = self.get()
+            template.metadata.resource_version = curr.metadata.resource_version
+        except NotFound:
+            return self.create()
+        return self.workflows_service.update_cluster_workflow_template(
+            self.name,
+            ClusterWorkflowTemplateUpdateRequest(template=template),
+        )
+
+    def lint(self) -> TWorkflow:
+        """Lints the ClusterWorkflowTemplate using the Argo cluster."""
+        assert self.workflows_service, "workflow service not initialized"
+        assert self.namespace, "workflow namespace not defined"
+        return self.workflows_service.lint_cluster_workflow_template(
+            ClusterWorkflowTemplateLintRequest(template=self.build())
+        )
+
+    def build(self) -> TWorkflow:
+        """Builds the ClusterWorkflowTemplate and its components into an Argo schema ClusterWorkflowTemplate object."""
+        # Note that ClusterWorkflowTemplates are exactly the same as WorkflowTemplates except for the kind which is
+        # handled in Workflow._set_kind (by __name__). When using ClusterWorkflowTemplates via templateRef, clients
+        # should specify cluster_scope=True, but that is an intrinsic property of ClusterWorkflowTemplates from our
+        # perspective.
+        return _ModelClusterWorkflowTemplate(**super().build().dict())
```

Comment:
Good catch, it has exactly the same `metadata: v1_1.ObjectMeta` member, so it will let you set the namespace, but I guess Argo will ignore it on submission. Would still be part of the metadata though so maybe we should validate/remove it?

Reply:
Added `_set_namespace` to raise an error if trying to set a value